inc/dashboard/namespace.php

```diff
@@ -0,0 +1,441 @@
+<?php
+/**
+ * Dashboard Analytics
+ *
+ * @package aws-analytics
+ */
+
+namespace Altis\Analytics\Dashboard;
+
+use function Altis\Analytics\Blocks\get_views as get_block_views;
+use function Altis\Analytics\Utils\query;
+
+/**
+ * Set up the Dashboard Analytics page.
+ */
+function setup() {
+	add_filter( 'manage_posts_columns', __NAMESPACE__ . '\\remove_default_columns', 10, 2 );
+	add_filter( 'post_row_actions', __NAMESPACE__ . '\\remove_post_row_actions', 10, 2 );
+	add_filter( 'manage_edit-xb_sortable_columns', __NAMESPACE__ . '\\xb_table_sorting' );
+	add_filter( 'request', __NAMESPACE__ . '\\xb_block_column_orderby' );
+	add_filter( 'bulk_actions-edit-xb', '__return_empty_array' );
+	add_filter( 'views_edit-xb', __NAMESPACE__ . '\\render_date_range_links' );
+	add_filter( 'months_dropdown_results', '__return_empty_array' );
+	add_filter( 'manage_xb_posts_columns', __NAMESPACE__ . '\\add_microcopy_to_column_titles', 99 );
+
+	add_action( 'pre_get_posts', __NAMESPACE__ . '\\modify_views_list_query' );
+	add_action( 'admin_enqueue_scripts', __NAMESPACE__ . '\\enqueue_styles' );
+}
+
+/**
+ * Enqueue styles.
+ */
+function enqueue_styles() {
+	wp_enqueue_style( 'xb-insights', plugin_dir_url( __FILE__ ) . '/assets/style.css', [], '18-03-2021', 'screen' );
+}
+
+/**
+ * Remove the default post list columns.
+ *
+ * @param array $columns An array of columns to filter.
+ * @param string $post_type The current post type.
+ *
+ * @return array The filtered columns array.
+ */
+function remove_default_columns( $columns, $post_type ) : array {
+	if ( 'xb' === $post_type ) {
+		unset( $columns['cb'] );
+		unset( $columns['title'] );
+		unset( $columns['altis_publication_checklist_status'] );
+	}
+
+	return $columns;
+}
+
+/**
+ * Remove the post quick actions.
+ *
+ * @param array $actions An array of actions that can be done on a single post.
+ * @param object $post The WP_Post object
+ *
+ * @return array The filtered array (empty for xbs).
+ */
+function remove_post_row_actions( $actions, $post ) : array {
+	if ( 'xb' === $post->post_type ) {
+		unset( $actions['edit'] );
```

Comment:
Might be useful to have a link to edit the post the XB is on? Though I think if we're removing the default title column these aren't shown anyway.

Reply:
they're are shown anyway -- I think they display by default for post list tables unless you explicitly remove them (like I did). Editing the original post from the list seems like a good idea, but what if the same block is used in multiple places (we don't surface that now, but we've discussed maybe doing it in the future)? Might be easier not to lock that in and then have to change it later